lua/nvim-treesitter/parsers.lua

```diff
@@ -58,15 +58,12 @@ end
 ---@field readme_name string|nil
 
 ---@type ParserInfo[]
-local list = {}
-if vim.fn.has "nvim-0.11" == 0 then
-  setmetatable(list, {
-    __newindex = function(table, parsername, parserconfig)
-      rawset(table, parsername, parserconfig)
-      ts.language.register(parsername, parserconfig.filetype or parsername)
-    end,
-  })
-end
+local list = setmetatable({}, {
+  __newindex = function(table, parsername, parserconfig)
+    rawset(table, parsername, parserconfig)
+    ts.language.register(parsername, parserconfig.filetype or (vim.fn.has "nvim-0.11" == 0 and parsername))
```

Comment:
```suggestion
    if parserconfig.filetype or vim.fn.has "nvim-0.11" == 0 then
      ts.language.register(parsername, parserconfig.filetype or parsername)
    end
```
Right now the right most expression gets evaluated as a boolean on nightly neovim, so this throws an error